### ISSUE TITLE
Transposed numpy array is written incorrectly

### ISSUE THREAD

**Issue by Ehtyar** [openPMD/openPMD-api#1882]
**Description of the bug**
When trying to save a transposed numpy array it is instead written to the file as if it was just reshaped instead.

**To Reproduce**
executable code example to reproduce the problem:

Python:
```python
import openpmd_api as io
import numpy as np

# have some 3D data that is supposed to be written
array = np.ndarray([some 3D data])

# make the series
series = io.Series(filepath, io.Access.create)
mesh = series.iterations[0].meshes["E"]["x"]

# the array is being transposed
data = np.transpose(array)

# then it is saved and the series is flushed and closed
dataset = io.Dataset(data.dtype, data.shape)
mesh.reset_dataset(dataset)
mesh[:, :, :] = data
series.flush()
series.close()
```

This writes the same thing to the disk as if I had written 

```python
data = np.reshape(array, array.shape[::-1])
```

This weirdly fragments the data.

@franzpoeschel @PrometheusPi

**Comment by Ehtyar** [comment]
Workaround:

force numpy to actually adjust the memory state, not just the indexing:
```python
data1 = np.transpose(array)
data = np.zeros(data1.shape, dtype=data1.dtype)
data[:,:,:]=data1[:,:,:]
```
This works at least but it isn't pretty.

**Comment by franzpoeschel** [comment]
I will have a look next week. From first glance this looks as if the only thing we can really do here is catch this condition and throw an error. Otherwise we would have to implement the entire protocol for operations that numpy has already enqueued in metadata but not realized in memory yet (openPMD operates on the raw Numpy buffer, which in terms of the transposed matrix would be strided at this point)

**Comment by franzpoeschel** [comment]
While our Python bindings check for contiguous buffers, this particular use case currently sidesteps the check because our check accidentally also allows Fortran-style buffers. Since transposing the buffer creates a contiguous Fortran-style buffer, our Python bindings silently accept this and then continue with the behavior you observed.

As we do not support writing from / reading into strided buffers (as created by transposing), the correct fix on our end is to reject the transposed buffer just like any other non-contiguous buffer. 
With the fix in #1886, your Python snippet explicitly fails instead of silently accepting a noncontiguous input:
```
> python ~/git-repos/openPMD-api/examples/10_streaming_write.py 
Traceback (most recent call last):
  File "/home/franzpoeschel/git-repos/openPMD-api/examples/10_streaming_write.py", line 23, in <module>
    mesh[:, :, :] = data
    ~~~~^^^^^^^^^
IndexError: strides in chunk are inefficient, not implemented!
```

So, you will still need the workaround (i.e. you need to prepare a contiguous buffer before storing).

Thank you for reporting
